libsolidity/ast/TypeProvider.cpp

```diff
@@ -233,7 +233,11 @@ Type const* TypeProvider::fromElementaryTypeName(ElementaryTypeNameToken const&
 	case Token::UFixed:
 		return fixedPoint(128, 18, FixedPointType::Modifier::Unsigned);
 	case Token::Address:
+	{
+		if (_stateMutability && *_stateMutability == StateMutability::Payable)
```

Comment:
For the current use case that would work, but do we want that to be general?

Reply:
I think we should assert - if we want to allow other mutabilities at any point in the future, then we have to touch this again anyways, so better to make sure that we in fact do that by asserting... otherwise one might first miss it and silently get ``address`` for e.g. ``address view``.

Reply:
Ok